app/services/foreman/renderer/scope/macros/kickstart.rb

```diff
@@ -0,0 +1,118 @@
+          include Foreman::Renderer::Errors
+          extend ApipieDSL::Module
+
+          apipie :class, desc: 'Macros to use within a kickstart template' do
+            name 'Kickstart'
+            sections only: %w[all]
+          end
+
+          # For more information about KS network directive: https://pykickstart.readthedocs.io/en/latest/kickstart-docs.html#network
+          apipie :method, "Returns a kickstart 'network' directive for a specific interface" do
+            required :iface, Nic::Managed, "Managed interface object to represent"
+            keyword :rhel_compatible, [true, false], "Is the OS rhel-compatible?"
+            keyword :os_major, Integer, "OS major version number"
+            keyword :use_slaac, [true, false], "Is the interface configured using SLAAC"
+            keyword :static, [true, false], "Use static configuration for IPv4"
+            keyword :static6, [true, false], "Use static configuration for IPv6"
+            returns String, "'network' directive with all switches that represent the input NIC object"
+            example 'kickstart_network(iface, rhel_compatible: true, os_major: 10, use_slaac: false) #=> "network --bootproto=dhcp --device=ens3"'
+          end
+          def kickstart_network(iface, host:, rhel_compatible:, os_major:, use_slaac:, static:, static6:)
+            return nil unless iface.is_a?(Nic::Managed)
+
+            network_options = []
+            nameservers = []
+            subnet4 = iface.subnet
+            subnet6 = iface.subnet6
+
+            # device and hostname
+            if iface.bond? && rhel_compatible && os_major >= 6
+              network_options.push("--device=#{iface.identifier}")
+            else
+              network_options.push("--device=#{iface.mac || iface.identifier}")
+            end
+            network_options.push("--hostname #{@host.name}")
+
+            # single stack
+            if subnet4 && !subnet6
+              network_options.push("--noipv6")
+            elsif !subnet4 && subnet6
+              network_options.push("--noipv4")
+            end
+
+            # dual stack MTU check
+            raise("IPv4 and IPv6 subnets have different MTU") if subnet4 && subnet6 && subnet4.mtu.present? && subnet6.mtu.present? && subnet4.mtu != subnet6.mtu
+
+            # mtu method is taking into account both ipv4 and ipv6 stacks
+            network_options.push("--mtu=#{iface.mtu}") if iface.mtu
+
+            # IPv4
+            if (subnet4 && !subnet4.dhcp_boot_mode?) || static
+              network_options.push("--bootproto static")
+              network_options.push("--ip=#{iface.ip}")
+              network_options.push("--netmask=#{subnet4.mask}")
+              network_options.push("--gateway=#{subnet4.gateway}")
+            elsif subnet4&.dhcp_boot_mode?
+              network_options.push("--bootproto dhcp")
+            end
+            if subnet4
+              nameservers.concat(subnet4.dns_servers)
+            end
```

Comment:
If there is no IPv4 subnet, you can't use static IPv4 since it needs to read values from it.
```suggestion
            if subnet4
              if !subnet4.dhcp_boot_mode? || static
                network_options.push("--bootproto static")
                network_options.push("--ip=#{iface.ip}")
                network_options.push("--netmask=#{subnet4.mask}")
                network_options.push("--gateway=#{subnet4.gateway}")
              elsif subnet4.dhcp_boot_mode?
                network_options.push("--bootproto dhcp")
              end
              nameservers.concat(subnet4.dns_servers)
            end
```

Same below for `subnet6` and `static6`.